modules/peers.js

```diff
@@ -520,16 +589,28 @@ Peers.prototype.onPeersReady = function () {
 			},
 			updatePeers: function (seriesCb) {
 				var updated = 0;
-				var peers = library.logic.peers.list();
+				var peers = self.acceptable(library.logic.peers.list());
 
 				library.logger.trace('Updating peers', {count: peers.length});
 
 				async.each(peers, function (peer, eachCb) {
 					// If peer is not banned and not been updated during last 3 sec - ping
 					if (peer && peer.state > 0 && (!peer.updated || Date.now() - peer.updated > 3000)) {
 						library.logger.trace('Updating peer', peer);
-						self.ping(peer, function (err) {
-							++updated;
+						peer.rpc.status(function (err, status) {
```

Comment:
It repeats all the way - move to dedicated function or integrate with `peer.rpc.status`